implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/v2/DynatraceExporterV2.java

```diff
@@ -0,0 +1,362 @@
+            throwIfValueIsInvalid(total);
+
+            serializedLine.add(
+                    createMetricBuilder(meter, meter.getId().getName())
+                            .setDoubleSummaryValue(min, max, total, count)
+                            .serialize());
+        } catch (MetricException e) {
+            logger.warn(String.format("Could not serialize metric with name %s: %s", meter.getId().getName(), e.getMessage()));
+        } catch (IllegalArgumentException iae) {
+            logger.warn(String.format("Illegal value for metric with name %s: %s Dropping...", meter.getId().getName(), iae.getMessage()));
+        }
+
+        return streamOf(serializedLine);
+    }
+
+    Stream<String> toDistributionSummaryLine(DistributionSummary meter) {
+        HistogramSnapshot histogramSnapshot = meter.takeSnapshot();
+        double total = histogramSnapshot.total();
+        double max = histogramSnapshot.max();
+        return toSummaryLine(meter, histogramSnapshot, total, max, false);
+    }
+
+    Stream<String> toLongTaskTimerLine(LongTaskTimer meter) {
+        HistogramSnapshot histogramSnapshot = meter.takeSnapshot();
+        double total = histogramSnapshot.total(getBaseTimeUnit());
+        double max = histogramSnapshot.max(getBaseTimeUnit());
+
+        return toSummaryLine(meter, histogramSnapshot, total, max, true);
+    }
+
+    Stream<String> toTimeGaugeLine(TimeGauge meter) {
+        return toGauge(meter);
+    }
+
+    Stream<String> toFunctionCounterLine(FunctionCounter meter) {
+        return toCounter(meter);
+    }
+
+    Stream<String> toFunctionTimerLine(FunctionTimer meter) {
+        return toGauge(meter);
+    }
+
+    Stream<String> toMeterLine(Meter meter) {
+        return toGauge(meter);
+    }
+
+    private void throwIfValueIsInvalid(Double value) {
+        if (value.isNaN()) {
+            throw new IllegalArgumentException("Value cannot be NaN.");
+        }
+        if (value.isInfinite()) {
+            throw new IllegalArgumentException("Value cannot be infinite.");
+        }
+    }
+
+    Stream<String> toGauge(Meter meter) {
+        return streamOf(meter.measure()).map(
+                measurement -> {
+                    try {
+                        throwIfValueIsInvalid(measurement.getValue());
+                        String metricKey = createMetricKey(meter.getId().getName(),
+                                measurement.getStatistic().getTagValueRepresentation());
+                        if (metricKey == null) {
+                            return null;
+                        }
+                        return createMetricBuilder(meter, metricKey)
+                                .setDoubleGaugeValue(measurement.getValue())
+                                .serialize();
+                    } catch (MetricException e) {
+                        logger.warn(String.format("Could not serialize metric with name %s: %s", meter.getId().getName(), e.getMessage()));
```

Comment:
You could introduce constants for both format strings since they're used three times each.